.github/workflows/test-and-format.yml

```diff
@@ -0,0 +1,50 @@
+name: Test & Format
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [ main ]
+
+permissions:
+  id-token: write
+  contents: read
+  checks: write
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    # strategy:
+    #   matrix:
+    #     dotnet-version: [ '3.1.x', '6.0.x' ]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v3
```

Comment:
We have adopted the pattern of pinning the Action hash instead of the major/minor version to mitigate the risk of supply chain attacks in the pipeline dumping our repo secrets.

Reply:
Added